storage/filestore/store.go

```diff
@@ -79,7 +109,45 @@ func (store *Store) Create(ctx context.Context, ref storage.BlobRef, size int64)
 	if err != nil {
 		return nil, Error.Wrap(err)
 	}
-	return newBlobWriter(ref, store, file), nil
+	return newBlobWriter(ref, store, storage.MaxStorageFormatVersionSupported, file), nil
+}
+
+// SpaceUsed adds up the space used in all namespaces for blob storage
+func (store *Store) SpaceUsed(ctx context.Context) (space int64, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	var totalSpaceUsed int64
+	namespaces, err := store.GetAllNamespaces(ctx)
+	if err != nil {
+		return 0, Error.New("failed to enumerate namespaces: %v", err)
+	}
+	for _, namespace := range namespaces {
+		used, err := store.SpaceUsedInNamespace(ctx, namespace)
```

Comment:
Is there a chance to gain performance with concurrency here? Or would that cause problems?

Reply:
In this case, that might not cause problems, but then again it might. In Jessica's pending change, this will always happen in a side goroutine, out of line with normal request processing, and we will always want to avoid this goroutine taking up too much processing time or i/o time at the expense of the others.